config/src/test/java/org/springframework/security/config/web/server/OAuth2ResourceServerSpecTests.java

```diff
@@ -478,6 +479,59 @@ public void getJwtDecoderWhenNoBeansAndNoDslWiredThenWiringException() {
 		assertThatExceptionOfType(NoSuchBeanDefinitionException.class).isThrownBy(() -> jwt.getJwtDecoder());
 	}
 
+	@Test
+	public void getJwtAuthenticationConverterWhenBeanWiredAndDslWiredThenDslTakesPrecedence() {
+		GenericWebApplicationContext context = autowireWebServerGenericWebApplicationContext();
+		ServerHttpSecurity http = new ServerHttpSecurity();
+		http.setApplicationContext(context);
+		ReactiveJwtAuthenticationConverter beanWiredJwtAuthenticationConverter = new ReactiveJwtAuthenticationConverter();
+		ReactiveJwtAuthenticationConverter dslWiredJwtAuthenticationConverter = new ReactiveJwtAuthenticationConverter();
+		context.registerBean(ReactiveJwtAuthenticationConverter.class, () -> beanWiredJwtAuthenticationConverter);
+		ServerHttpSecurity.OAuth2ResourceServerSpec.JwtSpec jwt = http.oauth2ResourceServer().jwt();
+		jwt.jwtAuthenticationConverter(dslWiredJwtAuthenticationConverter);
+		assertThat(jwt.getJwtAuthenticationConverter()).isEqualTo(dslWiredJwtAuthenticationConverter);
+	}
+
+	@Test
+	public void getJwtAuthenticationConverterWhenTwoBeansWiredAndDslWiredThenDslTakesPrecedence() {
+		GenericWebApplicationContext context = autowireWebServerGenericWebApplicationContext();
+		ServerHttpSecurity http = new ServerHttpSecurity();
+		http.setApplicationContext(context);
+		ReactiveJwtAuthenticationConverter beanWiredJwtAuthenticationConverter = new ReactiveJwtAuthenticationConverter();
+		ReactiveJwtAuthenticationConverter dslWiredJwtAuthenticationConverter = new ReactiveJwtAuthenticationConverter();
+		context.registerBean("firstJwtAuthenticationConverter", ReactiveJwtAuthenticationConverter.class,
+				() -> beanWiredJwtAuthenticationConverter);
+		context.registerBean("secondJwtAuthenticationConverter", ReactiveJwtAuthenticationConverter.class,
+				() -> beanWiredJwtAuthenticationConverter);
+		ServerHttpSecurity.OAuth2ResourceServerSpec.JwtSpec jwt = http.oauth2ResourceServer().jwt();
+		jwt.jwtAuthenticationConverter(dslWiredJwtAuthenticationConverter);
+		assertThat(jwt.getJwtAuthenticationConverter()).isEqualTo(dslWiredJwtAuthenticationConverter);
+	}
+
+	@Test
+	public void getJwtAuthenticationConverterWhenTwoBeansWiredThenDefaultTakesPrecedence() {
```

Comment:
I'd expect this test to align with `getJwtDecoderWhenTwoBeansWiredThenThrowsWiringException` and `OAuth2ResourceServerConfigurerTests#getJwtAuthenticationConverterWhenDuplicateConverterBeansThenThrowsException`. Is there a reason that this diverges from those?

Reply:
That's an interesting question. The reason is that `ReactiveJwtAuthenticationConverter` is resolved through the `getBeanOrNull()` method defined in `ServerHttpSecurity`. If more than one bean is defined, the logic will use the default one and not throw an error. I based the implementation on how most of the beans for OAuth2 and OIDC are configured in that class (e.g. `OidcUserService` or `ReactiveClientRegistrationRepository`). You made me notice that, in the imperative version, we throw an error if there's more than one bean defined. I guess we want the same logic for the reactive version as well. Thoughts?

Reply:
Yes, that was my thinking. 

If an application declares more than one `ReactiveJwtAuthenticationConverter` bean, the application should clearly indicate which one should be used. An error is a simple way to do that.

Reply:
Perfect! I changed it to the agreed behaviour.